cli/lib/tasks/download.js

```diff
@@ -256,8 +251,11 @@ const downloadFromUrl = ({ url, downloadDestination, progress, ca }) => {
       // response headers
       started = new Date()
 
-      // if our status code does not start with 200
-      if (!/^2/.test(response.statusCode)) {
+      if (redirectVersion && redirectUrl) {
+        downloadFromUrl({ url: redirectUrl, progress, ca, downloadDestination, version: redirectVersion })
+        .then(resolve).catch(reject)
```

Comment:
We could set `followRedirect` to `false` above and handle all of redirects ourselves. Then we can avoid the floating non-const `redirectVersion` and `redirectUrl` variables. I like that it localizes your redirect changes to one place and allows the next dev to have less knowledge of what `request` is doing.

```suggestion
      // Handle redirects ourselves
      if (/^3/.test(response.statusCode)) {
        downloadFromUrl({ url: response.headers.location, progress, ca, downloadDestination, version: response.headers.location })
        .then(resolve).catch(reject)
```

Reply:
Hm.  In the clear light of morning, a good point.  Rearranged.